src/debug.c

```diff
@@ -1819,23 +1821,115 @@ void closeDirectLogFiledes(int fd) {
 }
 
 #ifdef HAVE_BACKTRACE
+#define BACKTRACE_MAX_SIZE 100
+#ifdef __linux__
+#include <sys/prctl.h>
+#include <sys/syscall.h>
+#include <dirent.h>
+
+static pid_t *get_ready_to_signal_threads_tids(pid_t pid, int sig_num, size_t *tids_len_output);
+
+#define MAX_BUFF_LENGTH 256
+typedef struct {
+    char thread_name[16];
+    int trace_size;
+    pid_t tid;
+    void *trace[BACKTRACE_MAX_SIZE];
+} stacktrace_data;
+
+static void *collect_stacktrace_data(void) {
+    /* allocate stacktrace_data struct */
+    stacktrace_data *trace_data = zmalloc(sizeof(stacktrace_data));
+
+    /* Get the stack trace first! */
+    trace_data->trace_size = backtrace(trace_data->trace, BACKTRACE_MAX_SIZE);
+    
+    /* get the thread name */
+    prctl(PR_GET_NAME, trace_data->thread_name);
+
+    /* get the thread id */
+    trace_data->tid = syscall(SYS_gettid);
```

Comment:
Why not call `gettid` directly?

Reply:
`gettid` is not signal safe as well :upside_down_face: 

Btw, probably it's better to go over the code again check each function carefully for signal safety. I see some more candidates that are not listed as signal safe: prctl, opendir, readdir, closedir